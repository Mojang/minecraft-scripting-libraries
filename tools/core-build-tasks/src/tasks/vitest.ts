import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
