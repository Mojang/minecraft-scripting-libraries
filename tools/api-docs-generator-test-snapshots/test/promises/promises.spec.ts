// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Promises', () => {
    it('Properly generates promises documentation and definitions', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
