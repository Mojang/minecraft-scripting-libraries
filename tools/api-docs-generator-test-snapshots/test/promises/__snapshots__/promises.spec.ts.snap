// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Promises > Properly generates promises documentation and definitions > # Of Files Generated 1`] = `4`;

exports[`Promises > Properly generates promises documentation and definitions > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: promise-module
    items:
    - name: promise-module
      href: promise-module/promise-module.md
    - name: ExampleClass
      href: promise-module/ExampleClass.md
"
`;

exports[`Promises > Properly generates promises documentation and definitions > msdocs/scriptapi/promise-module/ExampleClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: promise-module.ExampleClass Class
description: Contents of the promise-module.ExampleClass class.
---
# ExampleClass Class

## Methods
- [methodWhichTakesPromises](#methodwhichtakespromises)

### **methodWhichTakesPromises**
\`
methodWhichTakesPromises(promiseWithType: Promise<string>, promiseWithVoid: Promise<void>): Promise<void>
\`

#### **Parameters**
- **promiseWithType**: Promise&lt;*string*&gt;
- **promiseWithVoid**: Promise&lt;*void*&gt;

**Returns** Promise&lt;*void*&gt;
"
`;

exports[`Promises > Properly generates promises documentation and definitions > msdocs/scriptapi/promise-module/promise-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: promise-module Module
description: Contents of the promise-module module
---
# \`promise-module\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "promise-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [ExampleClass](ExampleClass.md)
"
`;

exports[`Promises > Properly generates promises documentation and definitions > typescript/promise-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "promise-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export class ExampleClass {
    private constructor();
    methodWhichTakesPromises(promiseWithType: Promise<string>, promiseWithVoid: Promise<void>): Promise<void>;
}
"
`;
